utils/version.go

```diff
@@ -0,0 +1,38 @@
+package utils
+
+import (
+	"fmt"
+
+	"github.com/drand/drand/protobuf/common"
+)
+
+const (
+	FallbackMayor = 0
+	FallbackMinor = 0
+	FallbackPatch = 0
+)
+
+type Version struct {
+	Mayor uint32
+	Minor uint32
+	Patch uint32
+}
+
+func (v Version) IsCompatible(verRcv Version) bool {
+	if verRcv.Mayor == FallbackMayor && verRcv.Minor == FallbackMinor && verRcv.Patch == FallbackPatch {
+		return true
+	}
+	if v.Mayor == verRcv.Mayor {
```

Comment:
This assumes now with interceptors that EVERYTHING that doesn't have the same major version is not compatible. Which is clearly untrue with the unchained vs chained version.

Reply:
I'm fine with this as the default. we should be strict and it's clearer forcing protocol breaking changes to happen at once.

That said, i don't see how this is going to extend to a group upgrading to a different protocol?  Do we have a plan for what sort of migration will look like, since this setup seems to expect that given node is speaking exactly one protocol / version to all other peers.

Reply:
@willscott, this versioning feature is strictly related to the node itself. If something changes at communication protocol (like gRPC or something like that), the major node version will be increased. This is not related to beacon protocol differences. We are applying [semver](https://semver.org) to the node communication protocol.